tests/test_gpu_examples.py

```diff
@@ -933,3 +936,49 @@ def test_causal_lm_training_mutli_gpu(self):
 
             # assert loss is not None
             self.assertIsNotNone(trainer.state.log_history[-1]["train_loss"])
+
+
+@require_torch_gpu
+class OffloadSaveTests(unittest.TestCase):
+    def setUp(self):
+        self.causal_lm_model_id = "gpt2"
+
+    def tearDown(self):
+        r"""
+        Efficient mechanism to free GPU memory after each test. Based on
+        https://github.com/huggingface/transformers/issues/21094
+        """
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    @pytest.mark.single_gpu_tests
+    @require_torch_gpu
+    def test_offload_merge(self):
+        r"""
+        Test merging and unloading of a model with offloaded modules.
+        """
+        tokenizer = AutoTokenizer.from_pretrained(self.causal_lm_model_id)
+        config = AutoConfig.from_pretrained(self.causal_lm_model_id)
+        with init_empty_weights():
+            model = AutoModelForCausalLM.from_config(config)
+
+        memory_limits = {0: "0.5GIB", "cpu": "5GIB"}
+        # offloads around half of all transformer modules
+        device_map = infer_auto_device_map(model, max_memory=memory_limits)
+        assert 0 in device_map.values()
```

Comment:
Could you please always use `self.assertX` for consistency?

Reply:
Changed to assertTrue